grovel/grovel.lisp

```diff
@@ -354,6 +354,12 @@ int main(int argc, char**argv) {
   (dotimes (i (length c-names))
     (format out "~&#endif~%")))
 
+(define-grovel-syntax feature (lisp-feature-name c-name)
+  (c-section-header out "feature" lisp-feature-name)
+  (format out "~&#ifdef ~A~%" c-name)
+  (c-format out "(cl:pushnew ~S cl:*features*)" lisp-feature-name)
+  (format out "~&#endif~%"))
```

Comment:
`cl:*features*` is quite convenient for `#+` and `#-`, but for some use cases perhaps a boolean variable/predicate would work better or an alternative list variable (with `*features*` as the default)? Did you consider any of these alternatives?

Reply:
Both of those suggestions sound like a good idea, especially the option to use a different lists. Both would be easy enough to add. How about splitting this into two forms?
``` lisp
;; for the list variant
(feature-list lisp-feature-name c-name &key (feature-list cl:*features*))
;; for the variable
(feature-var bool-name c-name)
```

Reply:
I think I'd be happy with just the first variant. Because features are fun to develop, hard to maintain and impossible to remove... 🙂 Having said that, perhaps we can fold the second variant into the existing `constant` clause?